web/src/main/java/org/openmrs/web/filter/GZIPRequestWrapper.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.filter;
+
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+
+import javax.servlet.ServletInputStream;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Wraps Response for GZipFilter
+ *
+ * @author Matt Raible, cmurphy@intechtual.com
+ */
+public class GZIPRequestWrapper extends HttpServletRequestWrapper {
+	
+	private static final Log log = LogFactory.getLog(GZIPResponseWrapper.class);
+	
+	protected HttpServletRequest origRequest = null;
+	
+	protected ServletInputStream stream = null;
+	
+	protected PrintWriter writer = null;
+	
+	protected int error = 0;
+	
+	public GZIPRequestWrapper(HttpServletRequest request) {
+		super(request);
+		origRequest = request;
+	}
+	
+	public ServletInputStream createInputStream() throws IOException {
+		return (new GZIPRequestStream(origRequest));
+	}
+	
+	public void finishRequest() {
+		try {
+			if (writer != null) {
+				writer.close();
+			} else {
+				if (stream != null) {
+					stream.close();
+				}
+			}
+		}
+		catch (IOException e) {}
+	}
+	
+	public ServletInputStream getInputStream() throws IOException {
```

Comment:
Any methods that override should be annotated with @Override. Otherwise I can't quickly say if it is correct or not.